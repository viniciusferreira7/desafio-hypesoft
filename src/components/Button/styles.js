import styled, { css } from 'styled-components';

export const Container = styled.button`
${({ theme, background, sizes }) => css`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 1.2rem ${theme.spacings.large};
  letter-spacing: 0.02rem;
  text-transform: uppercase;
  background:${background ? theme.colors.secondaryColor : theme.colors.white};
  color:${background ? theme.colors.white : theme.colors.primaryColor};
  font-size: ${sizes}rem;
  border-radius: 5px;
  border: none;
  transition: all 500ms ease-in-out;
  cursor: pointer;

&:hover {
  background:${!background ? theme.colors.secondaryColor : theme.colors.white};
  color:${!background ? theme.colors.white : theme.colors.primaryColor};
}
`}
`;
