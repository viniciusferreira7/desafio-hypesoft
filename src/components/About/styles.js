import styled, { css } from 'styled-components';
import { Container as Button } from '../Button/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
${() => css`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 2.8rem;
  grid-template-areas:'SrcImg' 'Description';
  padding-top: 10rem;
`}
`;

export const SrcImg = styled.div`
${({ theme, srcImg }) => css`
  grid-area: 'SrcImg';
  max-width: 56rem;
  width: 100%;
  height: 54.2rem;
  background-image: url(${srcImg});
  background-size: cover;
  border-radius: 1rem;
  box-shadow: 3.2rem 3.2rem   ${theme.colors.white};

  @media ${theme.media.lteMedium}{
    display: none;
  }
`}
`;

export const Description = styled.div`
${() => css`
  grid-area: 'Description';
  max-width: 67.2rem;

  > ${Button} {
    width: 16.2rem;
    height: 5.6rem;

    > img{
      margin-left: 1.1rem;
    }
  }
`}
`;

export const InfoHeading = styled.div`
${() => css`
`}
`;

export const InfoContainer = styled.div`
${({ theme }) => css`
  max-width: 67.2rem;
  display: flex;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    padding-bottom: 3.2rem;
  }
`}
`;

export const Group = styled.div`
${() => css`
  max-width: 20.6rem;
  display: flex;
  align-items: center;
  margin-right: 5rem;

  > ${TextContainer}:nth-child(2){
    max-width: 9.5rem;
    margin-left: 1.6rem;
    padding: 0;
  }
`}
`;

`}`;

export const GroupTwo = styled.div`
${() => css`
  width: 25.1rem;
  display: flex;
  align-items: center;

  > ${TextContainer}:nth-child(2){
    max-width: 9.5rem;
    margin-left: 1.6rem;
    padding: 0;
  }
`}
`;

`}`;
