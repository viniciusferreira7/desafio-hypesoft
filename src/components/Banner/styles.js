import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Button } from '../Button/styles';
import { Container as GridDot } from '../GridDot/styles';

export const Container = styled.section`
${({ theme }) => css`
  width: 100%;
  max-height:31.2rem;
  overflow: hidden;
  padding: 6.4rem 2%;
  position: relative;
  margin-top: 16.4rem;

  @media ${theme.media.lteMedium}{
      max-height:auto;
      padding-top: 3rem
    }

 >  ${GridDot}{
    opacity: 0.15;
    position: absolute;
    top: 6.4rem;
    left: 12.2rem;
  }


  ${Heading}{
      width:50rem;
      height: 9.6rem;
      margin: 0 auto;


      @media ${theme.media.lteMedium}{
        width: 100%;
        text-align: center;
        height: 15rem;
        margin-bottom: 7rem;
      }

      > span{
        color:#bc199c;
      }

      > span::before{
        display: none;
      }
    }

  > ${Button} {
      width: 17.7rem;
      height: 5.6rem;
      margin: 0 auto;
      margin-top: 7.2rem;

      > a{
        margin-right: 0.9rem;
      }
  }
`}
`;

export const SmallEllipse = styled.div`
  ${({ theme }) => css`
    width:7.6rem;
    height:7.6rem;
    border-radius: calc(7.6rem / 2);
    background: ${theme.colors.white};
    opacity: 0.03;
    position: absolute;
    top: 2.6rem;
    right: 11.6rem;

    @media ${theme.media.lteMedium}{
      top: 8rem;
      right: 0;
    }
  `}
`;

export const HugeEllipse = styled.div`
  ${({ theme }) => css`
    width:52.9rem;
    height:52.9rem;
    border-radius: calc(52.9rem / 2);
    background: ${theme.colors.white};
    opacity: 0.03;
    position: absolute;
    top: 13.9rem;
    right: 90.7rem;

    @media ${theme.media.lteMedium}{
      width:30rem;
      height:30rem;
      top: 0;
      right: 4rem;
    }
  `}
`;

export const ContainerGridDot = styled.div`
  ${() => css`
    position: absolute;
    top:15rem;
    right: 21rem;
    transform: rotate(-90deg);

    ${GridDot}{
      opacity: 0.3;

    }
  `}
`;
