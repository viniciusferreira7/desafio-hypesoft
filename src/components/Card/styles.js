import styled, { css } from 'styled-components';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
${({ theme }) => css`
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.4), rgba(223, 227, 235, 0.4)), #FFFFFF;
  border-radius: 6px;
  width: 31.2rem;
  max-height: 35.1rem;
  transition: all 400ms ease-in-out;

  @media ${theme.media.lteMedium}{
    width: 18.5rem;
    padding-bottom: 2%;
    max-height: 40rem;

    > svg {
      width:2.5rem !important;
      height:2.5rem !important;
      margin: 0;
    }

    ${TextContainer}{
      text-align: center;
      font-size: 2rem;
      padding:2.2rem 0 3.2rem 0 !important ;

      > strong {
        margin-top: 1.6rem;
      }
    }

  }

    > svg{
      width:4rem;
      height:4rem;
      margin: 2.2rem 2.2rem 0 2.2rem;
      color: #242424;
    }

    ${TextContainer}{
      padding:2.2rem 0 3.2rem 2.8rem ;
      position: relative;

      > strong {
        width: 6.4rem;
        height: 0.3rem;
        background-color: #ED1CA6;
        margin-top: 1.6rem;
        margin-bottom: 2.8rem;
        position: absolute;
        bottom: -1.3rem;
        left:6rem ;
      }
    }

    > img{
      margin: 2.8rem 0 3.6rem 3.5rem;
    }


    &:hover{
      background: #FFFFFF;
      box-shadow: 0px 20px 48px rgba(10, 25, 49, 0.09);
      border-radius: 6px;
      border: 0.3rem solid #ED1CA6;

      ${TextContainer}{
        color:#ED1CA6 ;
      }

      > svg {
        color: #ED1CA6;
      }
    }

;
`}
`;
