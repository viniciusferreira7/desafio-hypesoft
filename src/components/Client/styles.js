import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
${() => css`
  display: flex;
  align-items: center;
  width: 31.2rem;
  height: 10rem;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3), rgba(223, 227, 235, 0.3)), #FFFFFF;
  border-radius: 0.4rem;
`}
`;

export const Photo = styled.div`
${({ theme, srcImg }) => css`
  width: 5.2rem;
  height: 5.2rem;
  border-radius: calc(5.2rem / 2);
  background-image:url(${srcImg});
  background-size: cover;
  background-color: ${theme.colors.gray_500};
  margin: 2.4rem;
  `}
`;

export const Info = styled.div`
${() => css`
  ${TextComponent}{
    text-transform: capitalize;
    padding: 0;
    padding-bottom: 0.6rem;

    > span{
      color: #ED1CA6
;
    }
  }
`}
`;
