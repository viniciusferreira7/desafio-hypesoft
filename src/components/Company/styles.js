import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  padding-top: 20rem;
  justify-content: space-around;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    text-align: center;
  }

  ${Heading}{
    width: 312px;

    > strong{
      color:#bc199c;
      font-weight: 400;
    }
  }
`}
`;

export const ContainerLogo = styled.div`
${({ theme }) => css`
  display: flex;
  max-width: 93.6rem;
  max-height: 12.0rem;
  justify-content: space-around;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    width: 100%;
    max-height: 96.6rem;
  }
`}
`;
export const Logo = styled.div`
${({ theme }) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 168px;
  height: 120px;
  margin-right: 2.4rem;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3), rgba(223, 227, 235, 0.3)), #FFFFFF;
  border-radius: 0.8rem;
  transition: all 500ms ease-in-out;

  @media ${theme.media.lteMedium}{
    margin: 0 auto;
    margin-top: 3rem;
  }

  &:hover{
    border:1rem solid ${theme.colors.white};
  }

  >img{
    width: 11.304rem;
    height: 2.6rem;
  }

  >img:nth-of-type(5){
    width:2.756rem !important;
    height:1.205rem !important;
  }

;

`}
`;
