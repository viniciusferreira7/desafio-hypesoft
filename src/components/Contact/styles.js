import styled, { css } from 'styled-components';
import { Container as GridDot } from '../GridDot/styles';

export const Container = styled.section`
${({ theme }) => css`
  width: 100%;
  display: flex;
  padding: 10rem;
  align-items: center;
  margin-top: 22.7rem;
  position: relative;

  ${GridDot}{
    position: absolute;
    top: 16%;
    right:12%;
    transform: rotate(-90deg);
    opacity: 0.3;

    @media ${theme.media.lteMedium}{
      display: none;
  }
  }

  @media ${theme.media.lteMedium}{
    padding: 2rem;
    flex-direction: column;
  }
`}`;
