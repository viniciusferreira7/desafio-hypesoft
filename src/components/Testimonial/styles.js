import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Client } from '../Client/styles';

export const Container = styled.div`
${({ theme }) => css`
  padding-top: 22rem;

  @media ${theme.media.lteMedium}{
    padding-top: 3rem;
  }


> ${Heading}{
    position: relative;
    text-align: center;
    text-transform: capitalize;

    > div {
      width: 8rem;
      height: 0.3rem;
      background-color: #ED1CA6;
      margin-top: 1.6rem;
      position: absolute;
      left:48% ;

      @media ${theme.media.lteMedium}{
        left:38%;
      }
    }
  }
`}
`;

export const ClientContainer = styled.div`
${({ theme }) => css`
  width: 100%;
  padding-top: 5rem;
  display: flex;
  align-items: flex-start;

  @media ${theme.media.lteMedium}{
        flex-direction: column;
      }
`}
`;

export const Group = styled.div`
${({ theme }) => css`
  margin-right: 4.42rem;


  @media ${theme.media.lteMedium}{
    margin: 0 auto;
    margin-bottom: 3rem;
  }

> ${Client}{

}

`}
`;
