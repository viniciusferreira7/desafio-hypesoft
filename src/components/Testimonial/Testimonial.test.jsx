import { fireEvent, screen } from '@testing-library/react';
import { Testimonial } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<Testimonial/>', () => {
  it('should render with default values', () => {
    renderTheme(<Testimonial title="test" />);
    expect(screen.getByRole('heading', { name: 'test' })).toBeInTheDocument();
  });

  it('should render correctly comments', () => {
    renderTheme(<Testimonial title="test" />);
    expect(screen.getByText(/looking/i)).toBeInTheDocument();

    fireEvent.click(screen.getByText(/cleber/i).parentElement);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Testimonial title="test" />);
    expect(container).toMatchSnapshot();
  });
});
