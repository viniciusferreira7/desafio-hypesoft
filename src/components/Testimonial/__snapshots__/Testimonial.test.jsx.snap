// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Testimonial/> should match snapshot 1`] = `
.c2 {
  color: #ffff;
  font-weight: 400;
  font-size: 4.0rem;
}

.c2 > span {
  color: #7D00FF;
  position: relative;
  z-index: 2;
}

.c2 > span::before {
  content: '';
  opacity: 0.2;
  position: absolute;
  z-index: 1;
  top: 60%;
  left: 0;
  width: 90%;
  height: 3rem;
  background: #35016b;
}

.c11 {
  font-size: 1.8rem;
  color: #242424;
  padding: 3.2rem 0;
  line-height: 2.8rem;
}

.c12 {
  font-size: 1.4rem;
  color: #A5A5A5;
  padding: 3.2rem 0;
  line-height: 2rem;
}

.c18 {
  font-size: 3.2rem;
  color: #242424;
  padding: 3.2rem 0;
  line-height: 5.12rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 31.2rem;
  height: 10rem;
  background: linear-gradient(0deg,rgba(223,227,235,0.3),rgba(223,227,235,0.3)),#FFFFFF;
  background-position: center;
  border-radius: 0.4rem;
  margin-bottom: 0.375rem;
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
  overflow-x: auto;
  overflow-y: hidden;
  cursor: pointer;
}

.c5:active {
  background: #242424;
}

.c5:hover {
  border: 0.3rem solid #ED1CA6;
  margin-bottom: 1rem;
}

.c5:hover .c6 {
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
  border: 0.05rem solid #ED1CA6;
}

.c5:hover .c8 {
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
}

.c5:hover .c8 > .c10 {
  color: #ED1CA6;
}

.c7 {
  width: 5.2rem;
  height: 5.2rem;
  border-radius: calc(5.2rem / 2);
  background-image: url(assets/images/photo/cleber-photo.jpg);
  background-size: cover;
  background-color: #A5A5A5;
  margin: 2.4rem;
}

.c13 {
  width: 5.2rem;
  height: 5.2rem;
  border-radius: calc(5.2rem / 2);
  background-image: url(assets/images/photo/gustavo-photo.jpg);
  background-size: cover;
  background-color: #A5A5A5;
  margin: 2.4rem;
}

.c9 .c10 {
  text-transform: capitalize;
  padding: 0;
  padding-bottom: 0.6rem;
}

.c9 .c10 > span {
  color: #ED1CA6;
}

.c0 {
  padding-top: 22rem;
}

.c0 > .c1 {
  position: relative;
  text-align: center;
  text-transform: capitalize;
}

.c0 > .c1 > div {
  width: 8rem;
  height: 0.3rem;
  background-color: #ED1CA6;
  margin-top: 1.6rem;
  position: absolute;
  left: 48%;
}

.c3 {
  width: 100%;
  padding-top: 5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c4 {
  margin-right: 4.42rem;
}

.c14 {
  background: linear-gradient(0deg,rgba(223,227,235,0.3),rgba(223,227,235,0.3)),#FFFFFF;
  width: 96.4rem;
  min-height: 42.1rem;
}

.c14 > .c10 {
  max-width: 86.4rem;
  padding: 2.6rem 5rem 5rem 5rem;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2.4rem 2.42rem 0 5.27rem;
  position: relative;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 > img {
  width: 2.667rem;
  height: 2.536rem;
  margin-right: 1.067rem;
}

.c17 {
  position: absolute;
  right: 2.42rem;
}

@media (max-width:768px) {
  .c5 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c0 {
    padding-top: 3rem;
  }
}

@media (max-width:768px) {
  .c0 > .c1 > div {
    left: 38%;
  }
}

@media (max-width:768px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c4 {
    margin: 0 auto;
    margin-bottom: 3rem;
  }
}

@media (max-width:768px) {
  .c14 {
    width: 100%;
    height: auto;
    padding: 0.5rem 1%;
  }
}

@media (max-width:768px) {
  .c14 > .c10 {
    width: 100%;
    font-size: 1.5rem;
    line-height: 2rem;
    padding-top: 1rem;
    height: auto;
  }
}

@media (max-width:768px) {
  .c15 {
    padding-top: 1rem;
  }
}

@media (max-width:768px) {
  .c17 {
    right: 5rem;
  }

  .c17 >img {
    width: 2.0rem;
    height: 4.0rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1 c2"
      color="#ffff"
    >
      test
       
      <div />
    </h1>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
          id="0"
        >
          <div
            class="c6 c7"
          />
          <div
            class="c8 c9"
          >
            <p
              class="c10 c11"
              color="#242424"
              sizes="1.8"
            >
              Cleber Souza
            </p>
            <p
              class="c10 c12"
              color="#A5A5A5"
              sizes="1.4"
            >
              Data Analyst
               of 
              <span>
                Bonfiglioli
              </span>
            </p>
          </div>
        </div>
        <div
          class="c5"
          id="1"
        >
          <div
            class="c6 c13"
          />
          <div
            class="c8 c9"
          >
            <p
              class="c10 c11"
              color="#242424"
              sizes="1.8"
            >
              Gustavo Borges
            </p>
            <p
              class="c10 c12"
              color="#A5A5A5"
              sizes="1.4"
            >
              Technical Design
               of 
              <span>
                Air Liquide Brasil 
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="c14"
      >
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <img
              alt="start"
              src="assets/images/start.png"
            />
            <p
              class="c10 c12"
              color="#A5A5A5"
              sizes="1.4"
            >
              5
               Star Rating
            </p>
          </div>
          <div
            class="c17"
          >
            <img
              alt="qoute 1"
              src="assets/images/qoute.png"
            />
            <img
              alt="qoute"
              src="assets/images/qoute.png"
            />
          </div>
        </div>
        <p
          class="c10 c18"
          color="#242424"
          sizes="3.2"
        >
          “If you're looking for someone who will challenge your UX/UI thinking and really cut to the core of what's important for users, then Jesse is your man. On top of that, he brings a level of enthusiasm to the craft that's energizing for everyone who works with him.”
        </p>
      </div>
    </div>
  </div>
</div>
`;
